src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -149,7 +156,7 @@ private Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectRest
                     SolutionRestoreRequest.OnUpdate(),
                     token);
 
-                return restoreTask;
+                return await restoreTask;
```

Comment:
This makes me sad haha. I've tried so many times to avoid us doing it here :D

Reply:
I could not only ignore your other comment, but go the other way and make this class less async  :)